src/main/js/field/FieldsPanel.js

```diff
@@ -104,8 +130,10 @@ class FieldsPanel extends React.Component {
 
         fields.forEach((field) => {
             const fieldKey = field.key;
-            const newField = FieldMapping.createField(field, currentConfig, fieldErrors[fieldKey], this.handleChange);
-            createdFields.push(newField);
+            if (!this.state.hiddenFieldKeys.includes(fieldKey)) {
```

Comment:
You might want to look into creating an isHiddenField function since you use this.state.hiddenFieldKeys.includes a few times might make it a little more readable.

Reply:
At the moment I believe it's the only place that uses this functionality. I could add a function for it now, but it'd contain just one line and that seems strange to me if we're not reusing it yet.

Reply:
Ok sounds reasonable to me.  I wasn't overly concerned about it.